src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -0,0 +1,330 @@
+                foreach (var gitFile in gitFiles)
+                {
+                    var fileInfo = new FileInfo(gitFile);
+                    fileInfo.Attributes = FileAttributes.Normal;
+                    fileInfo.IsReadOnly = false;
+                }
+                fileSystem.DeleteDirectory(tmpPath, true);
+            }
+        }
+        return 0;
+    }
+
+    private async Task<(NativePath tmpProductRepo, NativePath tmpVmrProductRepo, string mapping)> PrepareReposAsync(DiffRepo productRepo, DiffRepo vmr, NativePath tmpPath)
+    {
+        var tmpProductRepo = await PrepareProductRepoAsync(productRepo, tmpPath);
+        var mapping = versionDetailsParser.ParseVersionDetailsFile(tmpProductRepo / VersionFiles.VersionDetailsXml).Source?.Mapping
+            ?? Path.GetFileName(tmpProductRepo);
+        var tmpVmrProductRepo = await PrepareVmrAsync(vmr, tmpPath, mapping);
+
+        return (tmpProductRepo, tmpVmrProductRepo, mapping);
+    }
+
+    private async Task<NativePath> PrepareVmrAsync(DiffRepo vmr, NativePath tmpPath, string mapping)
+    {
+        if (string.IsNullOrEmpty(mapping))
+        {
+            throw new ArgumentException($"When preparing VMR, mapping can't be null");
+        }
+
+        var vmrProductRepo = tmpPath / Guid.NewGuid().ToString();
+        if (vmr.IsLocal)
+        {
+            await CheckoutBranchAsync(vmr);
+            fileSystem.CopyDirectory(Path.Combine(vmr.Path, VmrInfo.SourceDirName, mapping), vmrProductRepo, true);
+        }
+        else
+        {
+            vmrProductRepo = await PartiallyCloneVmrAsync(tmpPath, vmr, mapping);
+        }
+        await GitInitRepoAsync(vmrProductRepo, vmr.Branch);
+
+        return vmrProductRepo;
+    }
+
+    private async Task<NativePath> PrepareProductRepoAsync(DiffRepo repo, NativePath tmpPath)
+    {
+        var tmpProductRepo = tmpPath / Path.GetFileName(repo.Path);
+
+        if (!repo.IsLocal)
+        {
+            await processManager.ExecuteGit(Path.GetDirectoryName(tmpProductRepo)!, [
+                "clone",
+                "--depth", "1",
+                repo.Path,
+                "-b", repo.Branch,
+                tmpProductRepo
+            ]);
+        }
+        else
+        {
+            await CheckoutBranchAsync(repo);
+            fileSystem.CopyDirectory(repo.Path, tmpProductRepo, true);
+        }
+
+        return tmpProductRepo;
+    }
+
+    private async Task<(DiffRepo repo1, DiffRepo repo2)> ParseInput()
+    {
+        DiffRepo repo1, repo2;
+        var parts = options.Repositories.Split("..", StringSplitOptions.RemoveEmptyEntries);
+        if (parts.Length > 2 || parts.Length < 1)
+        {
+            throw new ArgumentException($"Invalid input {options.Repositories}. Input should be in the following format:" +
+                "repoPath/Uri:branch..repoPath/Uri:Branch or repoPath/Uri:branch when called from a git repo");
```

Comment:
```suggestion
                "remote:branch..remote:branch or remote:branch when called from a git repo");
```

`repoPath/Uri` is confusing

Reply:
true, remote is much better